samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -147,8 +161,17 @@ public ClusterBasedJobCoordinator(Config coordinatorSystemConfig) {
 
     metrics = new MetricsRegistryMap();
 
-    //build a JobModelReader and perform partition assignments.
-    jobModelManager = buildJobModelManager(coordinatorSystemConfig, metrics);
+    // Initialize CoordinatorStreamSystem to share between JobModelManager and ChangelogPartitionManager
+    CoordinatorStreamSystemConsumer coordinatorSystemConsumer = new CoordinatorStreamSystemConsumer(coordinatorSystemConfig, metrics);
+    CoordinatorStreamSystemProducer coordinatorSystemProducer = new CoordinatorStreamSystemProducer(coordinatorSystemConfig, metrics);
+
+    //build a JobModelManager and ChangelogPartitionManager and perform partition assignments.
+    CoordinatorStream coordinatorStream = new CoordinatorStream(coordinatorSystemConfig, metrics, COORDINATOR_STREAM_SOURCE);
+    coordinatorStream.startConsumer();
```

Comment:
Should just be a coordinatorStream#start and stop.

Reply:
I think I originally did this because the `LocalityManager` in SamzaContainer was just using the producer, so I figured I'd give an option to start both or just one.